crates/teloxide-core/src/errors.rs

```diff
@@ -89,36 +88,102 @@ impl AsResponseParameters for crate::RequestError {
     }
 }
 
+macro_rules! match_prefix {
+    ("") => {{
+        |data: &str| Some(data.to_owned())
+    }};
+    ($prefix: literal) => {{
+        |data: &str| {
+            if data.starts_with($prefix) {
+                Some(data.to_owned())
+            } else {
+                None
+            }
+        }
+    }};
+}
+
+macro_rules! impl_api_error {
+    (
+    $(#[$meta: meta])*
+    $vis: vis enum $ident: ident {
+        $($(#[$var_meta: meta])*
+        $var_name: ident$(($var_inner: ty))? = $var_string: literal $(with $var_parser: block)?),* } ) => {
```

Comment:
```suggestion
    (
        $( #[$meta:meta] )*
        $vis:vis enum $ident:ident {
            $(
                $( #[$var_meta:meta] )*
                $var_name:ident $( ($var_inner:ty) )? = $var_string:literal $(with $var_parser: block)?
             ),*
         } 
    ) => {
```
IMO (his would be a bit more readable (and copy the same formatting to the expansion)